vpr/src/place/move_utils.cpp

```diff
@@ -520,6 +560,16 @@ ClusterBlockId propose_block_to_move(int& logical_blk_type_index, bool highly_cr
         }
     }
 
+    if (b_from) {
```

Comment:
This should be a subroutine I think (cleaner, less cognitive pollution for the main code path).
It should be guarded by your #ifdef VTR_ENABLE_DEBUG_LOGGING